### PR TITLE
Prepare to solc-typed-ast v16.0.0 release

### PR DESCRIPTION
**Pull request by blitz-1306**
## Changes
- [x] Update dependencies (including fixes for #231 and #234), use **solc-typed-ast** v16.0.0.
- [x] Adapt to use `InferType.isFunctionCallExternal()`.
- [x] Update test artifacts to respect `SPDX-License-Identifier`s at the beginning of source units (that are now written when AST is written to a code).

Regards.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,8 +18,8 @@
         "findup-sync": "^5.0.0",
         "fs-extra": "^11.1.1",
         "logplease": "^1.2.15",
-        "semver": "^7.5.1",
-        "solc-typed-ast": "^15.0.0",
+        "semver": "^7.5.4",
+        "solc-typed-ast": "file:../solc-typed-ast",
```
Comment (cd1m0) [inline package.json:22]: Just a reminder to fix before final approval

**Review by cd1m0 — CHANGES_REQUESTED** [review]
LGTM. Just leaving as "request changes" so we can wait for the solc-typed-ast release

File: package.json (hunk 1 above)
Comment (blitz-1306) [reply package.json]: Fixed in 412160a (v16 is released now).

**Review by cd1m0 — APPROVED** [review]
lgtm

**Pull request merged**